packages/venia-concept/src/RootComponents/Product/Product.js

```diff
@@ -60,6 +24,15 @@ class Product extends Component {
         window.scrollTo(0, 0);
     }
 
+    // map Magento 2.3.1 schema changes to v2.0.0 proptype shape to maintain backwards compatibility
+    mapProduct(product) {
```

Comment:
I talked it over with @tjwiebell so I understand why these functions exist but I don't really like them :P I would prefer if we could do this in a single place such as a normalization/transform layer. I can see this getting unwieldy.

Reply:
See https://github.com/magento-research/pwa-studio/pull/990#discussion_r265708897

Reply:
@sirugh I thought about that too, but Tommy is right that this is the right way to do it for a workaround. If we create an abstract mapping layer, we're already halfway down the path to an ORM.